pkg/skaffold/render/renderer/kpt/kpt.go

```diff
@@ -117,7 +111,20 @@ func (r *Kpt) Render(ctx context.Context, out io.Writer, builds []graph.Artifact
 		cmd := exec.Command("kpt", "fn", "render", p, "-o", "unwrap")
 
 		if buf, err := util.RunCmdOut(rCtx, cmd); err == nil {
-			manifestList.Append(buf)
+			reader := kio.ByteReader{Reader: bytes.NewBuffer(buf)}
+			b := bytes.NewBuffer([]byte{})
+			writer := kio.ByteWriter{Writer: b}
+			// Kpt fn render outputs Kptfile content in result, we don't want this in our manifestList as Kptfile resource cannot be deployed to k8s cluster.
+			pipeline := kio.Pipeline{Filters: []kio.Filter{framework.ResourceMatcherFunc(func(node *yaml.RNode) bool {
+				return node.GetKind() != kptfile.KptFileKind
```

Comment:
If I recall correctly, I think the issue is not only with the `Kptfile` but with all the yaml files during rendering: kpt will read all the yaml files in the folder and render them. So, in the case we have a folder structure like this: 
```
project/
  Kptfile
  deployment.yaml
  skaffold.yaml
```
We'll need to filter the `Kptfile` and the `skaffold.yaml` or a deployer like `kubectl` will fail; I think the kpt deployer is able to filter this by itself properly. 🤔

Reply:
users can also ignore skaffold.yaml file  by  .krmignore file, but there might be another problem related to ``fn render `` output, the command currently also output configMap file which cannot be used for deployment either, I'm gonna create an issue with kpt maintainers.

Reply:
We don't filter out skaffold.yaml ourselves,  any yamls with ``Apiversion``,  ``Kind``, ``metadata.name`` will be included regardless of which kpt version we use .   It is users responsibility to make sure they put the right files in their kpt package  that should include ``Kptfile``, ``deployments files``, ``data config files`` , we can filter deployments files from the ``kpt fn render``.  Every standard kpt Data config file is expected to have ``config.kubernetes.io/local-config: "true"`` annotation , we can inspect the annotation for each yaml resource from ``kpt fn render`` and then filter them out.